src/typechecker.rs

```diff
@@ -21,7 +21,7 @@ use std::{
 /// The [`CheckedType`] enum contains three possible states about the type. Either the
 /// type has been properly resolved to something, or it corresponds to a Void type. If the
 /// type has not been resolved yet, it can be unknown.
-#[derive(Clone, PartialEq, Eq, Debug)]
+#[derive(Clone, PartialEq, Debug)]
```

Comment:
The nice window that we see here however is trigged by the `clippy.yml` and `actions-rs/clippy-check` action. So both are important to keep imo

Reply:
I agree with the fact that we need to keep using the action in the `clippy.yml` file
But I don't think that we should use it a second time in the `build` workflow

Reply:
Of course I will remove this change before merging, this was just to show we had the annotations